src/lib/statistics/statistics_objects/top_k_as_generic_histogram.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "abstract_histogram.hpp"
+#include "generic_histogram.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+const size_t TOP_K_DEFAULT = 100;
+
+class Table;
+
+/**
+ *  Top K histogram. 
+ *  This histogram saves the TOP_K_DEFAULT most common values as single bins in a GenericHistogram and all Non-Top K values as bins in between.
+ *  The Histogram is created as follows:
+ *  Each Top K value is modeled as one bin with a height of its stored count.
+ *  One bin is created for all Non-Top K values between two Top K bins using a uniform distribution assumption. 
+ */
```

Comment:
Does that overlap with the documentation in the cpp?

Reply:
It does. We found the more verbose explanation helpful for readers that only read the header file, but we can remove LL 21 - 23, as they contain essentially redundant information.

Reply:
Documentation and understandability is good. Try to avoid too many repetitions, you could also point to explanations in other files.

Reply:
Thanks for the input, will do.